src/ir/from_ast/astconv.rs

```diff
@@ -268,6 +278,14 @@ impl<'ctx, 'prog> BuildCtx<'ctx, 'prog> {
             has_interface: interface_port.is_some(),
         };
         let idx = self.comp.add(e);
+
+        // If the component is expecting interface information and there is an interface port, add it.
+        if let (Some((name, _)), Some(src)) =
+            (interface_port, &mut self.comp.src_info)
```

Comment:
Why is this a mutable reference?

Reply:
We need to mutate src_info if we are adding info to it.